design-proposals/eim-standalone-managed-en.md

```diff
@@ -0,0 +1,236 @@
+# Design Proposal: Convert Standalone Edge Nodes to Managed Edge Nodes
+
+Author(s): Tomasz Osiński
+
+Last updated: 29.04.2025
+
+## Abstract
+
+A Customer Journey for Open Edge Platform assumes that customers can manually deploy a set of Standalone
+Edge Nodes (SEN) that can be onboarded to the Edge Orchestrator at later stage, once a customer is ready to scale their deployment.
+SENs are converted to managed Edge Nodes which, once onboarded, are fully owned by the Edge Orchestrator - customers
+can manage them (e.g., install clusters, applications or perform Day2 OS updates) through the Edge Orchestrator UI and API.
+
+The Customer Journey is as follows:
+
+1. A customer installs one or more Standalone Edge Nodes following the user guides.
+2. The customer uses the SEN to deploy K8s clusters, applications, etc.
+3. The customer decides to scale out their deployment and onboard the SENs to the Edge Orchestrator.
+4. Once SENs are onboarded, the customer starts to use the Edge Orchestrator to manage the SENs. 
+   The customer can now use the Edge Orchestrator to manage the SENs, including installing clusters, applications, and performing Day2 OS updates.
+5. The customer can now provision additional Edge Nodes via remote provisioning or manually create additional SENs and follow the same workflow to onboard them.
+
+This document describes the design of the onboarding process of Standalone Edge Nodes to managed Edge Nodes (step 3) to allow for all further steps. 
+
+> NOTE: Converting standalone K8s clusters and applications to managed clusters and applications is out of scope for this document. 
+
+## Proposal
+
+### Design goals
+
+This design aims at:
+
+- Providing a solution for onboarding Edge Nodes in a fully automated way, with minimal manual steps.
+- Keeping the solution OS-independent, i.e., the solution should work on any OS that is supported by the Edge Orchestrator.
+- Reducing the user interaction (i.e., logging into the OS, injecting USB sticks, etc.) with Edge Node machines.
+- Enabling the onboarding process at scale, i.e., the design should enable onboarding of multiple machines at once.
+
+The proposed solution is an MVP approach - some of the above goals may not be fully achieved in the first version of the design,
+but the design doesn't preclude achieving them in the future. 
+
+### Assumptions
+
+- Customers will drive the onboarding process from a local developer machine, with desktop, keyboard and mouse.
```

Comment:
Is this still true? Reading below it seems like the CLI is executed directly on the SEN

Reply:
Kind of yes - I still assume that users do SSH into ENs and there is no peripherals attached to ENs